playbooks/roles/airship-deploy-tempest/tasks/main.yml

```diff
@@ -0,0 +1,110 @@
+---
+# tasks file for airship-deploy-tempest
+- name: Load standard variables
+  include_vars: "{{ playbook_dir }}/../vars/common-vars.yml"
+  tags:
+    - always
+
+- name: Set site path
+  set_fact:
+    site_path: "{{ upstream_repos_clone_folder }}/airship/treasuremap/site/{{ socok8s_site_name }}"
+  tags:
+    - always
+
+- name: Ensure site directory for {{ socok8s_site_name }} exists
+  become: yes
+  file:
+    path: "{{ site_path }}"
+    state: directory
+
+- name: Ensure tempest chart directory exists
+  become: yes
+  file:
+    path: "{{ site_path }}/software/charts/osh/openstack-tempest"
+    state: directory
+
+- name: Get external network ID
+  shell: "openstack network show {{ openstack_external_network_name }} -f value -c id"
+  environment:
+    OS_CLOUD: openstack
+  register: tempest_test_public_network_id
+  tags:
+    - skip_ansible_lint
+
+- name: Get external subnet CIDR
+  shell: "openstack subnet show {{ openstack_external_subnet_name }} -f value -c cidr"
+  environment:
+    OS_CLOUD: openstack
+  register: openstack_external_subnet_cidr
+  tags:
+    - skip_ansible_lint
+
+- name: Check if subnet pool stack exists
+  shell: "openstack stack show heat-subnet-pool-deployment"
+  environment:
+    OS_CLOUD: openstack
+  register: subnet_pool_result
+  failed_when:
+    - subnet_pool_result.rc != 0
+    - "'not found' not in subnet_pool_result.stderr"
+  changed_when: false
+  tags:
+    - skip_ansible_lint
+
+- name: Setup subnet pool
+  shell: "{{ role_path }}/files/tempest_setup_subnet.sh"
```

Comment:
This is the path to the roles directory on localhost (where run.sh is run from); this will fail if localhost is not the deployer host:

```
TASK [airship-deploy-tempest : Setup subnet pool] *****************************************************************************************************************************************************************************************************************************
fatal: [jharker-beefy-deployer]: FAILED! => changed=true 
  cmd: /home/opensuse/socok8s/playbooks/roles/airship-deploy-tempest/files/tempest_setup_subnet.sh
  delta: '0:00:00.005243'
  end: '2019-06-03 21:08:14.130153'
  msg: non-zero return code
  rc: 127
  start: '2019-06-03 21:08:14.124910'
  stderr: '/bin/sh: /home/opensuse/socok8s/playbooks/roles/airship-deploy-tempest/files/tempest_setup_subnet.sh: No such file or directory'
  stderr_lines:
  - '/bin/sh: /home/opensuse/socok8s/playbooks/roles/airship-deploy-tempest/files/tempest_setup_subnet.sh: No such file or directory'
  stdout: ''
  stdout_lines: <omitted>
```

Reply:
I agree with the request, but it should wait until the tempest tests can all pass (after blacklisting), otherwise, CI will suffer as a result.

Reply:
@jgu17 it looks like we need to run an extra action (`run.sh test`) to trigger this code path, so this won't affect CI until we add the test action to the Jenkinsfile